site/app/controllers/PollController.php

```diff
@@ -429,9 +452,21 @@ public function openPoll(): RedirectResponse {
             $this->core->addErrorMessage("Invalid Poll ID");
             return new RedirectResponse($this->core->buildCourseUrl(['polls']));
         }
-        $poll->setOpen();
-        $em->flush();
 
+        $duration = $poll->getDuration();
+        if ($duration->format('%r%Y%m%d%H%i%s') !== '000000000000') {
```

Comment:
Is there a cleaner way to do this, potentially by comparing against a 0 date interval?

Reply:
I tried multiple ways and ended up reverting this and checking each duration member variable separately. That was the only way it seemed to work. This and comparing it against a 0-date interval didn't work.